src/test/java/bio/terra/service/upgrade/MigrateTests.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.service.upgrade;
+
+import bio.terra.common.category.Unit;
+import bio.terra.service.resourcemanagement.google.GoogleResourceConfiguration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@Category(Unit.class)
+@SpringBootTest
+@ActiveProfiles({"google", "test"})
+@AutoConfigureMockMvc
+public class MigrateTests {
+
+    @Autowired
+    private Migrate migrate;
+
+    @Autowired
+    private MigrateConfiguration migrateConfiguration;
+
+    @Autowired
+    private GoogleResourceConfiguration googleResourceConfiguration;
+
+    private String singleDataProject;
+    @Before
+    public void setup() throws Exception {
+        singleDataProject = googleResourceConfiguration.getSingleDataProjectId();
```

Comment:
Would using a mock here avoid the need to save and restore the state of this object?

Reply:
Ok - I took a stab at implementing some sort of variation on mocking and it allowed me to build out the test suite even more!